### PR TITLE
Improve lambda performance

### LINKED ISSUE #6: Check Lambda performance
Profile validation lambda. 
Invocations seem to take ~17sec average, which seems way too long.

(Perhaps investigate use of xRay...)

### PR DESCRIPTION
**Pull request by williamputraintan**
Context:
Need to make lambda return function in less than 30 seconds as API Gateway has a strict 30 second rule. Currently, large samplesheet check may not return in time, and could ended up as a network error at the frontend.

Action in this PR:
- API call to fetch metadata is now in batch, avoiding a lot of network call to portal API
- Replace .query() to .loc[] when searching df from a certain values. (Save significant time)
- Lambda memory is increase to 256MB
- Make concurrency between check without metadata and fetching metadata from API
- Split umccr_util layer into runtime and the util code layer itself. (runtime layer only contain library from `pip install`)

### CODE REVIEW

**Review by victorskl — APPROVED** [review]
Tick approve. But I would defer @reisingerf sharp eyes for any other comments and final. Hope, async approach work. Trust you! 😉 

Side note: not about this PR. Overall, code is a bit hard to follow, now. (Not ur fault!) Understood the complexity around there. At some point, we might want to improve structure a bit i.e. application logic, validation logic (domain), external systems dependencies, module/packaging, etc.

**Comment by reisingerf** [comment]
A few comments:

- smaller changes please. You list a few changes, some of which make a nice PR on their own and would encapsulate a single change, like the switch from multiple to a single metadata API calls, or the change of `query()` to `loc[]`,... 
- just because I said we need to improve the runtime does not mean you have to implement everything right away. Nice investigative work and I hope you learned a bit. As mentioned there are some other options we could take this..

The code/project structure could be improved, but that's a learning process and I am happy as it went.  I was hoping to have time for a more in-detail code review (also with Victor, as he's much better than me at this), but not sure we get there. I don't mind if we don't get to change the code, but I'd really like to discuss some strategies at least in theory.

Happy to trust you with this and move it along.

**Review by reisingerf — APPROVED** [review]

**Comment by williamputraintan** [comment]
> Smaller changes please
> Code/project structure could be improved

Noted!
